junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/ClassBasedTestDescriptor.java

```diff
@@ -421,6 +423,24 @@ private void invokeAfterAllCallbacks(JupiterEngineExecutionContext context) {
 				.forEach(extension -> throwableCollector.execute(() -> extension.afterAll(extensionContext)));
 	}
 
+	private void invokeTestInstancePreDestroyCallback(JupiterEngineExecutionContext context) {
+		if (isPerClassLifecycle(context)) {
```

Comment:
Maybe consider moving this `if` statement where you call the method instead of inside it

Reply:
good suggestion